homeassistant/components/generic_thermostat/climate.py

```diff
@@ -307,6 +307,11 @@ def _async_startup(*_):
         if not self._hvac_mode:
             self._hvac_mode = HVACMode.OFF
 
+    @property
+    def capability_attributes(self) -> dict[str, Any] | None:
```

Comment:
It seemed like things are migrating away from `extra_state_attributes` and to `capability_attributes`, which is why I put this here, but I wasn't totally sure.

Reply:
My understanding is that the `capability_attributes` don't change, but rather are defined as a part of the device structure. If this is the case, preset temperatures are not such capability attributes, but rather settings which can be embedded as attributes next to an entity.

Reply:
Yeah, as I said, I wasn't sure what belonged where. If they don't belong in `capability_attributes` do you know where a better place would be? Is it `extra_state_attributes` after all?

Reply:
Apparently, yes. The `device_state_attributes` is deprecated in favor of `extra_state_attributes`.

See [here](https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L373):

```
    @property
    def device_state_attributes(self) -> Mapping[str, Any] | None:
        """Return entity specific state attributes.

        This method is deprecated, platform classes should implement
        extra_state_attributes instead.
        """
        return None
```